plugins/docker/src/main/java/org/radargun/service/DockerService.java

```diff
@@ -0,0 +1,251 @@
+         dockerContainerId = createCmd.exec().getId();
+      }
+      log.infof("Created container %s with id %s", containerName, dockerContainerId);
+   }
+
+   private DockerClient configureDockerClient() {
+      DockerCmdExecFactory dockerCmdExecFactory = new NettyDockerCmdExecFactory();
+
+      DefaultDockerClientConfig.Builder configBuilder = DefaultDockerClientConfig
+         .createDefaultConfigBuilder();
+
+      DockerClientConfig dockerClientConfig = configBuilder
+         .withDockerHost(serverUri)
+         .withRegistryUrl(dockerRegistry)
+         .withRegistryUsername(username)
+         .withRegistryPassword(password)
+         .build();
+
+      return DockerClientBuilder.getInstance(dockerClientConfig)
+         .withDockerCmdExecFactory(dockerCmdExecFactory)
+         .build();
+   }
+
+   private void stopAndRemoveContainer(String container) {
+      try {
+         dockerClient.stopContainerCmd(container).exec();
+      } catch (NotModifiedException e1) {
+         // Container was already stopped
+      }
+      log.warnf("Removing container %s", container);
+      dockerClient.removeContainerCmd(container).exec();
+   }
+
+   private void pullImage(String imageName) {
+      PullImageCmd pullImageCmd = this.dockerClient.pullImageCmd(imageName);
+      if (dockerRegistry != null) {
+         pullImageCmd.withRegistry(dockerRegistry);
+      }
+      pullImageCmd.exec(new PullImageResultCallback()).awaitSuccess();
+   }
+
+   @Override
+   public void start() {
+      dockerClient.startContainerCmd(dockerContainerId).exec();
+      started = true;
+      log.infof("Started container with id %s", dockerContainerId);
+      printContainerLog(dockerContainerId);
+   }
+
+   private void printContainerLog(String containerId) {
+      LogCallback loggingCallback = new LogCallback();
+
+      dockerClient.logContainerCmd(containerId)
+         .withStdErr(true)
+         .withStdOut(true)
+         .withFollowStream(true)
+         .withTailAll()
+         .exec(loggingCallback);
+
+      try {
+         loggingCallback.awaitCompletion(3, TimeUnit.SECONDS);
+      } catch (InterruptedException e) {
+         e.printStackTrace();
+      }
+      log.infof("Container log:\n%s", loggingCallback.toString());
+   }
+
+   private static class LogCallback extends LogContainerResultCallback {
+      protected final StringBuilder log = new StringBuilder();
+
+      @Override
+      public void onNext(Frame frame) {
+         log.append(new String(frame.getPayload()));
+      }
+
+      @Override
+      public String toString() {
+         return log.toString();
+      }
+   }
+
+   @Override
+   public void stop() {
+      dockerClient.stopContainerCmd(dockerContainerId).exec();
+      started = false;
+      log.infof("Stopped container with id %s", dockerContainerId);
+   }
+
+   @Override
+   public boolean isRunning() {
+      return started;
+   }
+
+   private static class EnvsConverter implements Converter<List<String>> {
```

Comment:
The problem is that dockerClient.withEnv expects a list of strings where each value is "key=value". I'd need to further convert the map to this list of strings hence I wrote my own converter.